### PR TITLE
Add engineOpts: A json encoded dictionary of generic options, e.g. tu…

### PR DESCRIPTION
**Pull request by vondele**
…ning parameters, to be used to initialize the engine

### CODE REVIEW

**Comment by vondele** [comment]
For example, allows for running a tuning branch with specific options:
E.g. https://github.com/XInTheDark/Stockfish/commit/d83d6a4eeaa6f5efa711ade509ea3c0da8fad968 

Can be run as:
```
python matecheck.py --engine ./stockfish --epdFile matetrack.epd --nodes 1000000 --engineOpts '{"a1": 80, "a2": 30, "a3": 95, "a4": 138, "a5": 103, "a7": 1802, "a8": 1154, "a9": 409, "a10": 1323, "b1": 51, "b3": 125, "b4": 68, "b8": 700, "b5": 591, "b6": 2, "b7": 1862, "c1": 10, "c2": 1368, "c3": 1471, "c4": 708, "c5": 590, "c6": 295, "c7": 6, "c9": 14389, "c10": 21, "c11": 212, "d1": 2, "d2": 169, "d3": 73, "d4": 526, "e2": 13, "e3": 282, "e4": 262, "e5": 103, "e6": 139, "e7": 168, "e8": 2922, "e10": 42, "e11": 181, "e12": 31, "e13": 220, "e14": 12, "e15": 14, "f1": 47, "f2": 58, "f3": 120, "f4": 491, "f5": 110, "f6": 141, "f7": 299, "f8": 192, "f9": 143, "f10": 8, "g1": 3077, "g2": 4664, "g3_b": 14, "g4": 49, "h1": 23, "h2": 114, "h3": 176, "h4": 118, "h5": 8, "h6": 81, "h7": 102, "h8": 179, "h9": 107, "h11": 52, "h12": 61, "i1": 312, "i2": 1103, "i3": 72, "i4": 876, "i5": 908, "i6": 510, "i8": 52, "aCorrHist": 1372, "b2": 13424, "c8": 215, "c12": 316, "e9": 3713, "g3": 13825, "h10": 83, "h13": 85, "h14": 343}' --concurrency 30
Loaded 6554 FENs, with max(abs(bm)) = 126.
Additional generic engine options:  {'a1': 80, 'a2': 30, 'a3': 95, 'a4': 138, 'a5': 103, 'a7': 1802, 'a8': 1154, 'a9': 409, 'a10': 1323, 'b1': 51, 'b3': 125, 'b4': 68, 'b8': 700, 'b5': 591, 'b6': 2, 'b7': 1862, 'c1': 10, 'c2': 1368, 'c3': 1471, 'c4': 708, 'c5': 590, 'c6': 295, 'c7': 6, 'c9': 14389, 'c10': 21, 'c11': 212, 'd1': 2, 'd2': 169, 'd3': 73, 'd4': 526, 'e2': 13, 'e3': 282, 'e4': 262, 'e5': 103, 'e6': 139, 'e7': 168, 'e8': 2922, 'e10': 42, 'e11': 181, 'e12': 31, 'e13': 220, 'e14': 12, 'e15': 14, 'f1': 47, 'f2': 58, 'f3': 120, 'f4': 491, 'f5': 110, 'f6': 141, 'f7': 299, 'f8': 192, 'f9': 143, 'f10': 8, 'g1': 3077, 'g2': 4664, 'g3_b': 14, 'g4': 49, 'h1': 23, 'h2': 114, 'h3': 176, 'h4': 118, 'h5': 8, 'h6': 81, 'h7': 102, 'h8': 179, 'h9': 107, 'h11': 52, 'h12': 61, 'i1': 312, 'i2': 1103, 'i3': 72, 'i4': 876, 'i5': 908, 'i6': 510, 'i8': 52, 'aCorrHist': 1372, 'b2': 13424, 'c8': 215, 'c12': 316, 'e9': 3713, 'g3': 13825, 'h10': 83, 'h13': 85, 'h14': 343}

Matetrack started for ./stockfish on matetrack.epd with --nodes 1000000 ...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 122/122 [07:17<00:00,  3.59s/it]


Using ./stockfish on matetrack.epd with --nodes 1000000
Engine ID:     Stockfish dev-20240713-d83d6a4e
Total FENs:    6554
Found mates:   4275
Best mates:    3308


```

File: README.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ and is visualized in the graphs below.
 ### Usage of `matecheck.py`
 
 ```
-usage: matecheck.py [-h] [--engine ENGINE] [--nodes NODES] [--depth DEPTH] [--time TIME] [--mate MATE] [--hash HASH] [--threads THREADS] [--syzygyPath SYZYGYPATH] [--minTBscore MINTBSCORE] [--maxTBscore MAXTBSCORE] [--concurrency CONCURRENCY] [--epdFile EPDFILE [EPDFILE ...]] [--showAllIssues] [--shortTBPVonly] [--showAllStats] [--bench]
+usage: matecheck.py [-h] [--engine ENGINE] [--nodes NODES] [--depth DEPTH] [--time TIME] [--mate MATE] [--hash HASH] [--threads THREADS] [--syzygyPath SYZYGYPATH]
+                    [--minTBscore MINTBSCORE] [--maxTBscore MAXTBSCORE] [--concurrency CONCURRENCY] [--engineOpts ENGINEOPTS] [--epdFile EPDFILE [EPDFILE ...]] [--showAllIssues]
```
Comment (robertnurnberg) [inline README.md:22]: Could leave all the line wraps to gui?

File: README.md [hunk 2]
```diff
@@ -38,13 +41,16 @@ options:
   --maxTBscore MAXTBSCORE
                         highest cp score for a TB win: if nonzero, it is assumed that (MAXTBSCORE - |score|) is distance in plies to first zeroing move in(to) TB (default: 20000)
   --concurrency CONCURRENCY
-                        total number of threads script may use, default: cpu_count() (default: 8)
+                        total number of threads script may use, default: cpu_count() (default: 32)
+  --engineOpts ENGINEOPTS
+                        A json encoded dictionary of generic options, e.g. tuning parameters, to be used to initialize the engine (default: None)
```
Comment (robertnurnberg) [inline README.md:46]: Drop the A?

**Comment by robertnurnberg** [comment]
Cannot test for the next few weeks. But looks good to me.

**Pull request merged**